ocaml/xapi/message_forwarding.ml

```diff
@@ -3361,6 +3361,23 @@ functor
           (host_uuid ~__context host)
           value ;
         Local.Host.set_uefi_certificates ~__context ~host ~value
+
+      let host_sched_gran_to_string = function
+        | `core -> "core"
+        | `cpu -> "cpu"
+        | `socket -> "socket"
+
+      let set_sched_gran ~__context ~self ~sched =
+        info "Host.set_sched_gran: host='%s' sched='%s'"
+          (host_uuid ~__context self)
+          (host_sched_gran_to_string sched) ;
+        Local.Host.set_sched_gran ~__context ~self
```

Comment:
This needs to be forwarded to the proper host, as it is written it'll always run on the pool master as you've noticed.
Look at other functions in this module, e.g. `request_backup`.

Reply:
Thanks a lot! That was really helpful :)